### PR TITLE
add comment functionality to activity tab

### PR DESCRIPTION
**Pull request by sorbaugh**
Issue: https://github.com/nextcloud-gmbh/server/issues/521
Mockups: https://github.com/nextcloud/server/issues/658

### CODE REVIEW

File: src/components/Moment.vue [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<!-- TODO: Move to vue components -->
```
Comment (artonge) [inline src/components/Moment.vue:1]: TODO

File: src/components/Moment.vue
Comment (artonge) [inline src/components/Moment.vue:1]: Licence is missing

File: src/components/Moment.vue [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+<template>
+	<span class="live-relative-timestamp" :data-timestamp="timestamp * 1000" :title="title">{{ formatted }}</span>
+</template>
+
+<script>
+import moment from '@nextcloud/moment'
+
+export default {
+	name: 'Moment',
```
Comment (artonge) [inline src/components/Moment.vue:11]: Name is too generic

File: src/mixins/CommentMixin.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+			}
+		},
+
+		// CREATION
+		async onNewComment(message) {
+			this.loading = true
+			try {
+				//const newComment = await NewComment(this.commentsType, this.ressourceId, message)
+				const newComment = await NewComment('files', 3, message)
+				// this.logger.debug('New comment posted', { commentsType: this.commentsType, ressourceId: this.ressourceId, newComment })
```
Comment (artonge) [inline src/mixins/CommentMixin.js:104]: Debug ?

File: src/sidebar.js [hunk 4]
```diff
@@ -41,21 +41,17 @@ const activityTab = new OCA.Files.Sidebar.Tab({
...
 
 	async mount(el, fileInfo, context) {
-		// only load if needed
-		if (ActivityTabView === null) {
-			const { default: ActivityTab } = await import('./views/ActivityTab.vue')
-			ActivityTabView = ActivityTabView ?? Vue.extend(ActivityTab)
-		}
-		// destroy previous instance if available
+		const { default: ActivityTab } = await import(/* webpackPreload: true */ './views/ActivityTab.vue')
+		ActivityTabView = ActivityTabView ?? Vue.extend(ActivityTab)
```
Comment (artonge) [inline src/sidebar.js]: Won't this revert this optimisation?
https://github.com/nextcloud/activity/pull/1363

**Review by artonge — CHANGES_REQUESTED** [review]
As the Comments tab need to be displayed if the activity app is disabled, the current approach leads to a lot of duplicate code.

A better approach would be to create an API so that the Comments app can add content to the actvity stream.

**Comment by susnux** [comment]
Closing in favor of https://github.com/nextcloud/activity/pull/1416

**Pull request closed without merging**